voicevox_engine/synthesis_engine/synthesis_engine.py

```diff
@@ -478,3 +487,120 @@ def synthesis(self, query: AudioQuery, speaker_id: int):
             wave = numpy.array([wave, wave]).T
 
         return wave
+
+    def guided_synthesis(
+        self,
+        query: AudioQuery,
+        speaker_id: int,
+        audio_file: Optional[IO],
```

Comment:
[QUESTION] Why is the `audio_file` argument set to `Optional`?
I think an error will occur if `audio_file` is `None`.
https://docs.python.org/3/library/typing.html#typing.Optional